client/src/areas/party/JoinFormRenderer.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from "react";
+import TextField, {
+  TextFieldProps
+} from "@material-ui/core/TextField/TextField";
+import Typography from "@material-ui/core/Typography/Typography";
+import {
+  ButtonWithProgress,
+  IButtonWithProgressProps
+} from "../../common/components/ButtonWithProgress";
+import styled from "styled-components";
+import { PartyManager } from "./PartyManager";
+
+export interface IJoinInfo {
+  address?: string;
+  password?: string;
+  joinAddress?: string;
+}
+
+interface IJoinFormState extends IJoinInfo {
+  doLogin?: boolean;
+  renderRegisterDialog?: boolean;
+  isLoading?: boolean;
+  error?: string;
+  success?: boolean;
+  joinButtonText: string;
+}
+
+export class JoinFormRenderer extends React.Component<{}, IJoinFormState> {
+  public constructor(props: {}) {
+    super(props);
+    this.state = {
+      joinButtonText: "Party please!"
+    };
+  }
+
+  public render() {
+    return (
+      <RootContainer>
+        <Typography variant="h6">Enter your holygrail info to join</Typography>
+        <FormContainer>
+          <div>
+            <StyledTextField
+              label="Holy Grail address"
+              onChange={e => this.handleAddressChange(e)}
+              onKeyPress={e => this.onKeyPress(e)}
+            />
+          </div>
+        </FormContainer>
+
+        <JoinButtonContainer>
+          {this.state.error && (
+            <ErrorContainer>{this.state.error}</ErrorContainer>
+          )}
+          <JoinButtonWithProgressWrapper
+            isLoading={this.state.isLoading}
+            isDisabled={!this.state.joinAddress}
+            onClick={this.join}
+            text={this.state.joinButtonText}
+            secondIcon="check"
+            showSecondIcon={this.state.success}
+          />
+        </JoinButtonContainer>
```

Comment:
There should be an explanation when you clicked Party please (that you have to wait blabla)

Reply:
Fixed in 97b24a3